File: docs/source/_snippets/simulation/environment_setup/environment_models/solar_sail_radiation_pressure.cpp
std::string sourceBody = "Sun";

double area = 2.0;

std::function< double( const double ) > coneAngle = [ = ]( const double ){ return 0.25; };
std::function< double( const double ) > clockAngle = [ =  ]( const double ){ return 0.2; };

double frontEmissivityCoefficient = 0.4;
double backEmissivityCoefficient = 0.4;
double frontLambertianCoefficient = 0.4;
double backLambertianCoefficient = 0.4;
double reflectivityCoefficient = 0.3;
double specularReflectionCoefficient = 1.0;

std::vector< std::string > occultingBodies; occultingBodies.push_back( "Earth" );
std::string& centralBody = "Earth";

bodySettings[ "Vehicle" ]->radiationPressureSettings[ sourceBody ] = std::make_shared< SolarSailRadiationInterfaceSettings >(
    sourceBody, area, coneAngle, clockAngle, frontEmissivityCoefficient, backEmissivityCoefficient,
    frontLambertianCoefficient, backLambertianCoefficient, reflectivityCoefficient, specularReflectionCoefficient, 
    occultingBodies, centralBody );

File: docs/source/_snippets/simulation/environment_setup/environment_models/solar_sail_radiation_pressure.py
source_body = "Sun"

area = 2.0

def cone_angle(): return 0.25

def clock_angle(): return 0.2

front_emissivity_coefficient = 0.4
back_emissivity_coefficient = 0.4
front_lambertian_coefficient = 0.4
back_lambertian_coefficient = 0.4
reflectivity_coefficient = 0.3
specular_reflection_coefficient = 1.0

occulting_bodies = [ "Earth" ]
central_body = "Earth"

radiation_pressure_settings = environment_setup.radiation_pressure.solar_sail( source_body, area, cone_angle, clock_angle,
	front_emissivity_coefficient, back_emissivity_coefficient, front_lambertian_coefficient, back_lambertian_coefficient,
	reflectivity_coefficient, specular_reflection_coefficient, occulting_bodies, central_body)

environment_setup.add_radiation_pressure_interface( bodies, "Vehicle", radiation_pressure_settings )